src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -1,16 +1,22 @@
 package com.scottlogic;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
 
 public class AuthorPostSorter implements PostSorter {
 
     @Override
-    public List<UserPost> sort(List<UserPost> inputList) {
-        if(inputList != null){
-                inputList.sort(Comparator.comparing(UserPost::getAuthor));
+    public List<UserPost> sort(List<UserPost> inputList, SortOrder SortOrder) {
+        if (inputList != null) {
+            List<UserPost> copyInputList = new ArrayList<>(inputList);
+            copyInputList.sort(Comparator.comparing(UserPost::getAuthor));
+            if (SortOrder == com.scottlogic.SortOrder.DESC) {
+                copyInputList.sort(Collections.reverseOrder(Comparator.comparing(UserPost::getAuthor)));
+            }
+            return copyInputList;
         }
-        return inputList;
+        return null;
```

Comment:
In general having a code block inside an `if` statment where the outside just returns is better done by having the inverse condition. So using `if (inputList == null) {return null}` makes the code look a lot cleaner. It means that the condition and action are close together and the rest of the code doesn't end up half way across the screen